.github/workflows/deploy-onpremise.yml

```
@@ -0,0 +1,99 @@
+      - 'on-premise/**'
+  workflow_dispatch:  # 수동 실행 가능
+
+jobs:
+  deploy-user-service:
+    name: Deploy User Service
+    runs-on: self-hosted
+    if: contains(github.event.head_commit.modified, 'on-premise/core-user-service') || github.event_name == 'workflow_dispatch'
+    
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
```

actionlint (1.7.10)

[error] the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)